Haptics/app/src/main/java/com/example/android/haptics/samples/ui/expand/ExpandRoute.kt

```diff
@@ -0,0 +1,337 @@
+                                Alignment.Center
+                            )
+                        )
+                    }
+                    // Draw indicators if there is no ongoing transition and expanded.
+                    if (transitionData.isExpanded) {
+                        Text(
+                            stringResource(R.string.expand_screen_tap_to_minimize),
+                            Modifier
+                                .align(Alignment.Center)
+                                .offset(y = -((EXPAND_SHAPE_COLLAPSED_SIZE.value / 2).dp))
+                        )
+                        Box(
+                            modifier = Modifier
+                                .size(16.dp)
+                                .clip(CircleShape)
+                                .background(MaterialTheme.colors.secondary)
+                                .align(Alignment.Center)
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Hold the transition values for expanding between states.
+ */
+private data class TransitionData(
+    val size: Dp,
+    val thickness: Float,
+    val isCollapsed: Boolean, // Transition is complete and now collapsed.
+    val isExpanded: Boolean, // Transition is complete and now expanded.
+)
+
+/**
+ * Create a transition and return it's animation values for animating between expanding and
+ * collapsing.
+ */
+@Composable
+private fun updateTransitionData(
+    expandShapeState: ExpandShapeState,
+    animateToExpandedDuration: Int,
+    animateToCollapsedDuration: Int
+): TransitionData {
+    val transition = updateTransition(expandShapeState, label = "Transition between ExpandShapeState.Collapsed and Expanded.")
+    // For the expanding and collapsing animation, we use a donut-like shape making it larger and
+    // and less thick.
+    val size by transition.animateDp(
+        transitionSpec = {
+            when {
+                ExpandShapeState.Collapsed isTransitioningTo ExpandShapeState.Expanded ->
+                    tween(durationMillis = animateToExpandedDuration, easing = LinearEasing)
+                else -> tween(durationMillis = animateToCollapsedDuration, easing = LinearEasing)
+            }
+        }, label = "Transition size between ExpandShapeState.Collapsed and Expanded."
+    ) { state ->
+        when (state) {
+            ExpandShapeState.Collapsed -> EXPAND_SHAPE_COLLAPSED_SIZE
+            ExpandShapeState.Expanded -> EXPAND_SHAPE_EXPANDED_SIZE
+        }
+    }
+
+    val thickness by transition.animateFloat(
+        transitionSpec = {
+            when {
+                ExpandShapeState.Collapsed isTransitioningTo ExpandShapeState.Expanded ->
+                    tween(durationMillis = animateToExpandedDuration, easing = LinearEasing)
+                else ->
+                    tween(durationMillis = animateToCollapsedDuration, easing = LinearEasing)
+            }
```

Comment:
Could you please extract and reuse this, like you did for the Resist example?

Reply:
done